fitting/fitScanData.py

```diff
@@ -277,7 +277,10 @@ def fit(self, debug=False):
 
                     # Provide an initial guess
                     init_guess_p0 = self.calDAC2Q_m[vfat]*(8+stepN*8)+self.calDAC2Q_b[vfat]
-                    init_guess_p1 = abs(self.calDAC2Q_m[vfat]*rand)  #self.calDAC2Q_m[vfat] might be negative (e.g. VFAT3 case)
+                    if self.isVFAT3:
+                        init_guess_p1 = self.calDAC2Q_m[vfat]*(256)+self.calDAC2Q_b[vfat]+abs(self.calDAC2Q_m[vfat]*rand)  #self.calDAC2Q_m[vfat] might be negative (e.g. VFAT3 case)
```

Comment:
What's the rational here?

You're setting the initial guess for `[1]` which represents the scurve width (should be ENC).  This should be in units of fC.  

```python
(self.calDAC2Q_m[vfat]*(256)+self.calDAC2Q_b[vfat])+abs(self.calDAC2Q_m[vfat]*rand)
```

So the first term in paranthesis (I added just to make it easier to reference) is the minimum (maximum) charge in voltage step (current injection mode), this is not the ENC.  Then you're adding to it a slope term that is in fC which could more aptly represent the ENC.

So why for the initial guess have this first term?  While it might work in the few test cases I wonder if there is not a better approach.  Remind me the reason you identified for this failing?  It was performing an infinite loop but I don't remember the cause.

Maybe there is a better more physical way to solve this than adding what seems like an arbitrary charge amount, which may fail in the case of current injection mode (unsure...)

Reply:
As for the whole term you have put in parentheses, I took that from a couple lines down where this is the lower bound on the range for the fit: https://github.com/brentstone/gem-plotting-tools/blob/develop/fitting/fitScanData.py#L199.

Since this initial value is always positive, I simply added it to the lower bound to ensure that the first guess is within our set bounds. The reason for this failing is essentially that the lower bound is too high for the guess. The boolean ```fitValid``` is used to determine whether to enter a ```continue``` statement (enter when it is false), and so we enter the infinite loop since our starting point is not within the bounds.

Reply:
Admittedly I didn't second guess the choice of lower bound; I agree that we should maybe think of a better starting point for the fit. However, the converged values typically were closer to the lower bound than the upper, which this method is suitable for.

Reply:
Update the comment on this line to reflect what we have done, e.g. added the guess to the lower bound and then I'll approve this.